packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -0,0 +1,166 @@
+  IGetDidSiopSessionArgs,
+  IRegisterCustomApprovalForDidSiopArgs,
+  IRemoveCustomApprovalForDidSiopArgs,
+  IRemoveDidSiopSessionArgs,
+  IRequiredContext,
+  ISendDidSiopAuthenticationResponseArgs,
+  IVerifyDidSiopAuthenticationRequestUriArgs
+} from '../types/IDidAuthSiopOpAuthenticator'
+
+/**
+ * {@inheritDoc IDidAuthSiopOpAuthenticator}
+ */
+export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
+  readonly schema = schema.IDidAuthSiopOpAuthenticator
+  readonly methods: IDidAuthSiopOpAuthenticator = {
+    getDidSiopSession: this.getDidSiopSession.bind(this),
+    addDidSiopSession: this.addSessionForDidSiop.bind(this),
+    removeDidSiopSession: this.removeSessionForDidSiop.bind(this),
+    authenticateWithDidSiop: this.authenticateWithDidSiop.bind(this),
+    getDidSiopAuthenticationRequestFromRP: this.getDidSiopAuthenticationRequestFromRP.bind(this),
+    getDidSiopAuthenticationRequestDetails: this.getDidSiopAuthenticationRequestDetails.bind(this),
+    verifyDidSiopAuthenticationRequestURI: this.verifyDidSiopAuthenticationRequestURI.bind(this),
+    sendDidSiopAuthenticationResponse: this.sendDidSiopAuthenticationResponse.bind(this),
+    registerCustomApprovalForDidSiop: this.registerCustomApprovalForDidSiop.bind(this),
+    removeCustomApprovalForDidSiop: this.removeCustomApprovalForDidSiop.bind(this)
+  }
+
+  private readonly sessions: Record<string, OperatingPartySession>
+  private readonly customApprovals: Record<string, (verifiedAuthenticationRequest: VerifiedAuthenticationRequestWithJWT) => Promise<void>>
+
+  constructor(customApprovals?: Record<string, (verifiedAuthenticationRequest: VerifiedAuthenticationRequestWithJWT) => Promise<void>>) {
+    this.sessions = {}
+    this.customApprovals = customApprovals || {}
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.getSession} */
+  private async getDidSiopSession(
+      args: IGetDidSiopSessionArgs,
+      context: IRequiredContext
+  ): Promise<OperatingPartySession> {
+    if (this.sessions[args.sessionId] === undefined) {
+      return Promise.reject(new Error(`No session found for id: ${args.sessionId}`))
+    }
+
+    return this.sessions[args.sessionId]
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.addSessionForDidSiop} */
+  private async addSessionForDidSiop(
+      args: ICreateDidSiopSessionArgs,
+      context: IRequiredContext
+  ): Promise<OperatingPartySession> {
+    if (this.sessions[args.sessionId] !== undefined) {
+      return Promise.reject(new Error(`Session with id: ${args.sessionId} already present`))
+    }
+
+    const sessionId = args.sessionId
+    const session = new OperatingPartySession({sessionId, identifier: args.identifier, expiresIn: args.expiresIn, context})
+    await session.init()
+    this.sessions[sessionId] = session
+
+    return session
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.removeSessionForDidSiop} */
+  private async removeSessionForDidSiop(
```

Comment:
I know we are using DidSiop in the SIOP library a lot as well, but propose to just rename all of these methods to omit the Did part, as it adds nothing. So this one would become removeSessionForSiop

Reply:
changed the names to omit did